go/arrow/flight/flightsql/driver/utils.go

```diff
@@ -60,44 +60,135 @@ func (g grpcCredentials) RequireTransportSecurity() bool {
 func fromArrowType(arr arrow.Array, idx int) (interface{}, error) {
 	switch c := arr.(type) {
 	case *array.Boolean:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return c.Value(idx), nil
 	case *array.Float16:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return float64(c.Value(idx).Float32()), nil
 	case *array.Float32:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return float64(c.Value(idx)), nil
 	case *array.Float64:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
 		return c.Value(idx), nil
+	case *array.Decimal128:
+		if c.IsNull(idx) {
+			return nil, nil
+		}
+
+		dt, ok := arr.DataType().(*arrow.Decimal128Type)
+		if !ok {
+			return nil, fmt.Errorf("datatype %T not matching decimal128", arr.DataType())
+		}
```

Comment:
this check is unnecessary, there should never be a case where the datatype for a `Decimal128` is not a `Decimal128Type`